src/common/components/feed/feed.utils.ts

```diff
@@ -134,19 +143,83 @@ export function generateFeedCardAttributes(
   }
 }
 
-export function shouldAggregate(e1: FeedQueryEvent, e2: FeedQueryEvent, e3: FeedQueryEvent) {
+export const getAggregateProfiles = (aggregateEvent: AggregateEvent): User[] => {
+  if (aggregateEvent.eventsAggregated.length < 2) {
+    return [
+      {
+        address: aggregateEvent.walletAddress,
+        profile: aggregateEvent.profile,
+      },
+    ];
+  }
+  const users: User[] = [];
+  aggregateEvent.eventsAggregated.map((user) => {
+    if (!users?.find((u) => u?.address === user.walletAddress)) {
+      users.push({
+        address: user.walletAddress,
+        profile: user.profile,
+      });
+    } else {
+      // do nothing
+    }
+  });
+  return users as User[];
+};
+
+export const aggregateEventsTime = (events: FeedQueryEvent[]) => {
+  let avgTime = 0;
+  events.forEach((event) => {
+    const date = DateTime.fromISO(event.createdAt).toMillis();
+    avgTime += date;
+  });
+  avgTime = avgTime / events.length;
+  const avgDate = DateTime.fromMillis(avgTime);
+  return avgDate;
+};
+
+export function shouldAggregateFollows(e1: FeedQueryEvent, e2: FeedQueryEvent, e3: FeedQueryEvent) {
   // for now
 
   if (!e1 || !e2 || !e3) return false;
 
   return (
     e1.__typename === e2.__typename &&
     e2.__typename == e3.__typename &&
-    e1.walletAddress === e2.walletAddress &&
-    e2.walletAddress === e3.walletAddress
+    // e1.walletAddress === e2.walletAddress &&
+    // e2.walletAddress === e3.walletAddress &&
+    e1.__typename === 'FollowEvent' &&
+    e2.__typename === 'FollowEvent' &&
+    e3.__typename === 'FollowEvent'
   );
 }
 
+export const shouldAggregateSaleEvents = (
```

Comment:
Same comment as above. Why not any number of events?

Reply:
Hard coded to 3 for now since we only want to aggregate at 3 or above